enhancements/metadata/user-tag-management.md

```diff
@@ -0,0 +1,396 @@
+}
+
+type AWSMetadataSpec struct {
+	OpCode string `json:"opcode"`
+    ResourceTags map[string]string `json:"resourcetags"`
+}
+
+type OpCodeType string
+
+// Opcodes supported by controller
+const (
+	OpAdd OpCodeType = "add"
+	OpUpdate OpCodeType = "update"
+	OpDelete OpCodeType = "delete"
+)
+
+type CloudProviderConditionType string
+
+const (
+	// CloudProviderConditionApplied indicates tags add/update operation completion condition.
+    CloudProviderConditionApplied MetadataConditionType = "applied"
+	
+	// CloudProviderConditionApproved signifies the tags validation condition.
+    CloudProviderConditionApproved MetadataConditionType = "approved"
+	
+	// CloudProviderConditionPolicy indicates if there are any policy condition failures.
+    CloudProviderConditionPolicy MetadataConditionType = "policy"
+	
+	// CloudProviderConditionRequest indicates any operation condition.
+    CloudProviderConditionRequest MetadataConditionType = "request"
+)
+
+type CloudProviderStatus struct {
+    Type CloudProviderConditionType `json:"type"`
+    Status metav1.ConditionStatus `json:"status"`
+    LastTransitionTime *metav1.Time `json:"lastTransitionTime,omitempty"`
+    Reason string `json:"reason, omitempty"`
+    Message string `json:"message, omitempty"`
+}
+
+type MetadataStatusConditionType string
+
+// Extensible for cases when there are multiple cloud providers being 
+// referred to tag management.
+const (
+    MetadataConditionReady MetadataStatusConditionType = "ready"
+)
+
+type MetadataStatus struct {
+	// Type and status signify the condition type and success status.
+    Type MetadataStatusCondtitionType `json:"type"`
+    Status metav1.ConditionStatus `json:"status"`
+	
+	// LastTransitionTime provides the time for last operation handled by the controller.
+    LastTransitionTime *metav1.Time `json:"lastTransitionTime,omitempty"`
+	
+	// Reason gives the error codes in string for failures.
+    Reason string `json:"reason, omitempty"`
+	
+	// Message details the string error messages.
+    Message string `json:"message, omitempty"`
+	
+	// Resourcestags is the active list of tags applied to cloud resources.
+    ResourceTags map[string]string `json:"resourcetags"`
+}
+
+type ClassifierSpec struct {
+    Classifiers map[string]string `json:"classifiers"`
+}
+
+type ControllerConfig struct {
+    LimitPolicy LPolicy `json:"limit, omitempty"`
+	SyncEvery  *string  `json:"syncevery, omitempty"`
+}
+```
+
+### Risks and Mitigations
+
+1. Any user with permission to edit custom resource can influence tags on cloud resources with wide-scoped classifiers.
+
+### Drawbacks
+
+1. User intervention is required in some cases for failure resolution of opcode driven operations.
+2. HA is not supported for the controller.
+3. Override of global configurations specific to cloud provider are not supported.
+4. Pre-existing tags or tags added using external tools on cloud resources can be updated using the controller.
```

Comment:
Is this a drawback? IMO this is part of enforcing policy, I think that's what this project is creating right, a tag policy for the clusters, making sure it's correct seems important to me, even if that means overwriting external changes